packages/@aws-cdk/aws-iam/lib/policy.ts

```diff
@@ -96,7 +96,16 @@ export class Policy extends Resource implements IPolicy {
         // generatePolicyName will take the last 128 characters of the logical id since
         // policy names are limited to 128. the last 8 chars are a stack-unique hash, so
         // that shouod be sufficient to ensure uniqueness within a principal.
-        Lazy.stringValue({ produce: () => generatePolicyName(resource.logicalId) })
+        Lazy.stringValue({
+          produce: () => {
+            // as logicalId is itself a Token,
+            // resolve it first before passing to generatePolicyName()
```

Comment:
I feel it makes more sense to change generatePolicyName to return a token and hide these details in there.

Reply:
Changed.